src/runtime_src/core/edge/ps_kernels/built_in/instance_query/instance_query.c

```diff
@@ -0,0 +1,272 @@
+    auto value = stream_data.substr(pos + 1, stream_data.size());
+
+    // Remove any extra spaces
+    boost::algorithm::trim(key);
+    boost::algorithm::trim(value);
+    // Generate node for data
+    pt::ptree new_node;
+    if (is_dict)
+      pt.put(key, value);
+    else {
+      new_node.put("name", key);
+      new_node.put("value", value);
+      pt.push_back(std::make_pair("", new_node));
+    }
+  }
+}
+
+static void
+add_schema(pt::ptree &pt)
+{
+    pt::ptree schema_pt;
+    schema_pt.put("schema", "JSON");
+    schema_pt.put("major", "1");
+    schema_pt.put("minor", "0");
+    schema_pt.put("patch", "0");
+    // Add the schema node
+    pt.add_child("schema_version", schema_pt);
+}
+
+
+// If unknown entries are detected they will be added into the output
+// ptree as listed unless the nonmatches flag is set
+static void
+filter_ptree_contents(pt::ptree& output, const pt::ptree& input, std::map<std::string, std::string>& filter, const bool add_nonmatches = true)
+{
+  for (const auto& item : input) {
+    const auto& it = filter.find(item.first);
+    if (it != filter.end())
+      output.put(it->second, item.second.data());
+    else if(add_nonmatches)
+      output.put(item.first, item.second.data());
+  }
+}
+
+static void
+get_mem_info(pt::ptree &pt)
+{
+  pt::ptree mem_pt;
+  parse_file(mem_pt, "/proc/meminfo");
+
+  static std::map<std::string, std::string> name_map = {
+      { "MemTotal", "mem_total"},
+      { "MemFree", "mem_free"},
+      { "MemAvailable", "mem_available"}
+  };
+  filter_ptree_contents(pt, mem_pt, name_map, false);
+}
+
+static void
+get_os_release(pt::ptree &pt)
+{
+    pt::ptree os_pt;
+    struct utsname sysinfo;
+    if (!uname(&sysinfo)) {
+      os_pt.put("sysname",   sysinfo.sysname);
+      os_pt.put("release",   sysinfo.release);
+      os_pt.put("version",   sysinfo.version);
+      os_pt.put("machine",   sysinfo.machine);
+    }
+
+    // Extract a single node for the distribution data
+    std::ifstream ifs("/etc/os-release");
+    if (ifs.good()) {
+      pt::ptree opt;
+      pt::ini_parser::read_ini(ifs, opt);
+      std::string val = opt.get<std::string>("PRETTY_NAME", "");
+      if (val.length()) {
+        if ((val.front() == '"') && (val.back() == '"')) {
+          val.erase(0, 1);
+          val.erase(val.size()-1);
+        }
+        os_pt.put("distribution", val);
+      }
+    }
+    ifs.close();
+
+    std::string model("unknown");
+    std::ifstream stream("/proc/device-tree/model");
+    if (stream.good()) {
+      std::getline(stream, model);
+      stream.close();
+    }
+    os_pt.put("model", model);
+
+    os_pt.put("cores", std::thread::hardware_concurrency());
+    os_pt.put("address_space", (boost::format("0x%lx") % (sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGE_SIZE))).str());
+    get_mem_info(os_pt);
+    pt.add_child("os", os_pt);
+}
+
+static void
+get_instance_process_status( const std::string& pid,
+                    pt::ptree &pt)
```

Comment:
```suggestion
get_instance_process_status(const std::string& pid, pt::ptree &pt)
```
Same for others.  It's weird how the argument list was broken here, but not in much longer functions above.